src/Multipart/AbstractUploadManager.php

```diff
@@ -92,8 +93,11 @@ public function promise()
         if ($this->promise) {
             return $this->promise;
         }
+        $exception = $this->client->getApi()->getServiceName() === 's3'
```

Comment:
Agreed, I added documentation and changed this to the constructor in the latest commit :)